plugins/Admin/admin.js

```diff
@@ -48,4 +49,21 @@ exports.uptime = {
 		}
 		msg.channel.send("**Uptime**: " + timestr);
 	}
-}
\ No newline at end of file
+}
+
+exports.userRole = {
+    usage: "<@user> <@role>",
+    description: "Used to toggle @role of specified @user",
+    process: function(bot,msg,arg){
+        var args = arg.split(" ");
+        var user, role;
+        if(typeof args[0] != 'undefined' && typeof args[1] != 'undefined'){
+            try{
+                if(msg.mentions.members.first()){ console.log('User mention '+msg.mentions.members.first().id); user = msg.mentions.members.first(); }
+                if(msg.mentions.roles.first()){ console.log('Role mention '+msg.mentions.roles.first().id); roleid = msg.mentions.roles.first().id; }
+            }catch(err){ console.log('Error occured at: '+err+' Mentions.first() is undefined'); }
```

Comment:
oh man actually took me a second to understand whats happening here. these inline if/else/catch blocks make this really hard to read. 
so !userRole @user @role toggles userrole got it.

might want to put this behind some admin checks so that every user doesnt give himself admin?

Reply:
so it is on admin.js so there is a little protection, I could do a has permission on the user running the command, there is also the role hierarchy in discord it self, as long as the bot role is below the admin role it will not be able to add or remove from it or any other role above it. Definitely will look at the has permission admin or change role.

Reply:
changed userRole to userrole, added a check to see if the message sender has permission on the server of MANAGE_ROLES, and sends a DM if they don't.